unix/povconfig/syspovconfig.h

```diff
@@ -102,21 +102,25 @@ using boost::intrusive_ptr;
 
 #endif // STD_POV_TYPES_DECLARED
 
-// After Stroustrop in _The C++ Programming Language, 3rd Ed_ p. 88
+// After Stroustrup in _The C++ Programming Language, 3rd Ed_ p. 88
 #ifndef NULL
 const int NULL=0;
 #endif
 
 #define POV_DELETE_FILE(name)  unlink(name)
 
 #if defined (PATH_MAX)
-# define FILE_NAME_LENGTH   PATH_MAX
+    // Use the system's actual limit if known.
+    // NB: PATH_MAX is understood including a terminating NUL character.
+    #define POV_FILENAME_BUFFER_CHARS   (PATH_MAX-1)
 #elif defined (_POSIX_PATH_MAX)
-# define FILE_NAME_LENGTH   _POSIX_PATH_MAX
+    // Otherwise, use the most restrictive limit allowed by POSIX if defined.
+    // NB: _POSIX_PATH_MAX is understood including a terminating NUL character.
+    #define POV_FILENAME_BUFFER_CHARS   (_POSIX_PATH_MAX-1)
 #else
-# define FILE_NAME_LENGTH   200
+    // As a fallback, use an even more conservative limit.
+    #define POV_FILENAME_BUFFER_CHARS   199
 #endif
-#define MAX_PATH FILE_NAME_LENGTH  // FIXME: remove later
```

Comment:
`MAX_PATH` has been eliminated from the platform-neutral code. It is still used in Windows-specific code, but only to set up buffers for API calls that are guaranteed to never write more than `MAX_PATH` bytes to the buffer.